src/docfx/build/metadata/MetadataProvider.cs

```diff
@@ -48,7 +50,52 @@ public JObject GetMetadata(Document file, JObject yamlHeader = null)
             {
                 result.Merge(yamlHeader, JsonUtility.MergeSettings);
             }
-            return result;
+
+            var invalidNames = new List<string>();
```

Comment:
Generate one message per key. This aligns with the warning messages we generate from `JsonUtility` validation for yaml header.

Reply:
i would like to use one error message per file, which is more user-friendly.